documentation/src/main/asciidoc/topics/con_jvm_heap_memory_allocation.adoc

```diff
@@ -1,26 +1,100 @@
-====
+The scenarios outlined in the previous table considered the following factors when allocating JVM heap space:
+
+* Clustered cache mode
+* Number of segments
+** For example, a low number of segments might affect how a server distributes data across nodes.
+* Read or write operations
+* Rebalancing requirements
+** For example, a high number of threads might quickly run in parallel during a state transfer, but each thread operation might consume more memory. 
+* Scaling clusters
+* Synchronous or asynchronous replication
+
+Consider the previous list of items when determining the amount of memory and CPU requirements for your {brandname} deployment. 
+Most notable {brandname} operations that require higher CPU limits include rebalancing nodes after pod restarts, running indexing queries on data, and performing GC operations. 
+ 
+The following example calculation helps determine the amount JVM memory required for an {brandname} deployment:
+
+[source,options="nowrap",subs=”+attributes.+quotes”]
+----
+JVM memory = JVM code + JVM stack size + JVM heap space
+----
+
+Note the following elements from the previous calculation:
+* A JVM stack, also called a thread stack, is an area of JVM memory that stores JVM threads. 
+* JVM heap space is an area of JVM memory that stores data and temporary objects from an application. 
+
+Additionally, consider the following calculation that can help you determine a suitable JVM heap space, where `m1` represents the calculated JVM heap space in megabytes. 
+
+[source,options="nowrap",subs=”+attributes.+quotes”]
+----
+m1 = (int) ((x * y) / (p * S)) + 1
+----
+
+The following table outlines the elements in the previous calculation:
+
+[%autowidth,cols="1,1",stripes=even,%noheader]
+|===
+| Element | Description
+
+|`int` | Denotes the calculated value for `m1` as an integer. 
+| `x` | Represents the average size of a key-value pair. You can determine the value of `x` bu using the following calculator: `x =  Serialized size of key + Serialized size of value + Metadata`
+| `y` a| Represents the expected number of cache entries for each node. Consider the type of cache mode when determining a value for `y`. For example:
+
+* For distributed mode, determine the amount of owners per cache entry.
+* For the local and replicated modes, 
+
+Note: Ensure that the number of entries (H) does not exceed the expected JVM heap space (S). Otherwise, {brandname} might evict more cache entries to optimize JVM memory space
+
+| `p` | 
+Denotes a fractional value for the full heap space that Infinsipan uses to store cache entries. Typical starting value is p=0.5.
+
+Consider that the JVM, heap size, GC parameters, and the amount of data stored in JVM memory can affect the length of a GC operation. You might need to configure your GC implementation to facilitate these operations.
+
+| `S` | 
+Represents the expected JVM heap space for each node in a cluster. Value in megabytes.
+|===
```

Comment:
The formular might be harder to understand than explanations.
Also ISPN-14024 requested a better description why 50% of the heap can not be used for data.
This explanation is completely missed from my perspective, we should give this explanation:
- traffic to clients need to de/encode and marshall entries (throughput)
- queries buffer the results (query throughput)
- segments are partially copied/marshalled in case of state-transfer (down- upscaling or cluster instability)
- a bit space to allow the (well tuned) GC algorythm find dead objects and mark for removal (to prevent fullGC)
- prevent GC algorythm to move objects from eden->ten generation right before removed
For the GC stuff I would point to the JVM documentation as it would be too much to explain that complicated stuff here :)

Reply:
Thanks, @wfink . I'm not a fan of these calculations, because I do not know how official or precise they are for all scenarios. 

I would prefer general descriptive items that can help a user understand why allocating 50 % for data storage might impact data processing operations. 

I think the feedback was in relation to why one would need to allocate 50 % when it has no relation to storing cache entries. What might the breakdown be for resource activities in relation to this 50 % estimate.

Best I remove both calculations and the related text?

Reply:
Updated.